numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1192,139 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class CAR(Distribution):
+    r"""
+    The Conditional Autoregressive (CAR) distribution is a special case of the multivariate
+    normal in which the precision matrix is structured according to the adjacency matrix of
+    sites. The amount of autocorrelation between sites is controlled by :math:`\alpha`. The
+    distribution is a popular prior for areal spatial data.
+
+    :param float or ndarray loc: mean of the multivariate normal
+    :param float alpha: autoregression parameter. For most cases, the value should lie
+        between 0 (sites are independent, collapses to an iid multivariate normal) and
+        1 (perfect autocorrelation between sites), but the specification allows for negative
+        correlations.
+    :param float tau: positive precision for the multivariate normal
+    :param numpy.ndarray or scipy.sparse.csr_matrix W: symmetric adjacency matrix where 1
+        indicates adjacency between sites and 0 otherwise
+    :param bool whether to use a sparse form of W in calculations (must be True is W is a
+        scipy.sparse.csr_matrix)
+    """
+
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.open_interval(-1, 1),
+        "tau": constraints.positive,
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+    ]
+
+    def __init__(self, loc, alpha, tau, W, *, is_sparse=False, validate_args=None):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+
+        self.is_sparse = is_sparse
+
+        if self.is_sparse:
+            if not sparse.issparse(W):
+                assert isinstance(W, np.ndarray)
+            self.W = sparse.csr_matrix(W)
+        else:
+            assert not sparse.issparse(W)
+            self.W = W
+
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(loc)[:-1],
+            jnp.shape(alpha),
+            jnp.shape(tau),
+            jnp.shape(self.W)[:-2],
+        )
+        event_shape = jnp.shape(self.W)[-1:]
+        self.loc = promote_shapes(loc, shape=batch_shape + event_shape)
```

Comment:
I guess the error that you got is due to this line. You need
```
self.loc, = promote...
```

Reply:
What does the , do?

Reply:
the `promote_shapes` function returns a list with 1 element, that trick selects the first element of the list.

Reply:
ah nice I didn't realise that sort of syntax worked without `, _`. Good to know!